### ISSUE TITLE
RFDETRMedium vs RFDETRKeypointPreview detection mismatch

### ISSUE THREAD

**Issue by vmrnd** [roboflow/rf-detr#1145]
Thank you for the great work at releasing the RFDETRKeypointPreview

From this https://blog.roboflow.com/real-time-keypoint-detection-with-rf-detr/ , my understanding is that the keypoint head is built directly into the detection transformer? Is that correct? If that's the case, why are the detections mismatched between RFDETRMedium for instance and RFDETRKeypointPreview.

```python
from rfdetr import RFDETRKeypointPreview, RFDETRMedium
import cv2
import supervision as sv

model_kp = RFDETRKeypointPreview(device="mps", num_classes=1)
model_det = RFDETRMedium(device="mps")

frame = cv2.imread("frame.jpg")

# Keypoint model
kp = model_kp.predict(frame, threshold=0.5)
kp = kp[kp.class_id == 1]
print(f"Keypoint @0.5: {len(kp)} people, scores={kp.detection_confidence}")

# Detection model (same threshold)
det = model_det.predict(frame, threshold=0.5)
det = det[det.class_id == 1]
print(f"Detection @0.5: {len(det)} people, scores={det.confidence}")
```

Output

`Keypoint @0.5: 2 people, scores=[2.75443   0.8900233]
Detection @0.5: 8 people, scores=[0.8627667  0.84324574 0.84246427 0.83921593 0.8308797  0.7930519
 0.7452643  0.52044135]`

Also I get the following warnings. Could this be why?

`[2026-06-21 12:48:39] [WARNING] rf-detr - Pretrained weights at '/Users/vmrnd/.roboflow/models/rf-detr-keypoint-preview-xlarge.pth' loaded only partially — this typically produces lower accuracy. 4 checkpoint key(s) not consumed by model: [keypoint_head.keypoint_proj.0.weight, keypoint_head.keypoint_proj.0.bias, keypoint_head.keypoint_proj.2.weight, keypoint_head.keypoint_proj.2.bias]. Check that the model configuration (encoder, hidden_dim, out_feature_indexes, projector_scale, ...) matches the architecture the checkpoint was trained with.`

**Comment by isaacrob** [comment]
That warning looks like a bug!

The model is inserted directly into the detector but they're both finetuned together and then optimized via NAS so neither the weights nor the detector part of the architecture match any of the existing detector models

**Comment by Borda** [comment]
About the warning, it is due to the removed MLP head, and I will fix it.
About the thresholding...
  - RFDETRMedium threshold=0.5 filters by normal object confidence.
  - RFDETRKeypointPreview threshold=0.5 filters by fused keypoint score.
  - Some duplicate/uncertain query boxes around people can have a fused score > 0.5.
  - If we filtered keypoint outputs by raw object confidence instead, some of those extra keypoint detections would likely fall below the threshold.
